noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/components/private_kernel_circuit_output_validator.nr

```diff
@@ -304,28 +284,26 @@ impl PrivateKernelCircuitOutputValidator {
             previous_kernel.end.unencrypted_logs_hashes,
             array_lengths.unencrypted_logs_hashes
         );
+        validate_array_prepended(
```

Comment:
Unfortunately this has increased 4k gate counts, contributing most part of the increment of init (3471) and inner (6489).
I think there's some unreasonably heavy stuffs going on and we should be able to reduce it a bit. (Will refactor in another pr!)